File: Assignments/Acme-Electronics-Shop/CustomerApp/app/src/main/java/pt/up/feup/cpm/customerapp/models/Customer.kt
package pt.up.feup.cpm.customerapp.models

import java.io.Serializable

data class Customer(
    var userID: String? = null,
    var name: String? = null,
    var address: String? = null,
    var fiscalNumber: String? = null,
    var email: String? = null,
    var password: String? = null,
    var cardType: String? = null,
    var cardNumber: String? = null,
    var cardValidity: String? = null,
    var publicKey: String? = null
): Serializable

File: Assignments/Acme-Electronics-Shop/Remote Service/models/Customer.js
import mongoose from "mongoose";

export const customerSchema = new mongoose.Schema({
    userID: {
        type: String,
        required: true
    },
    name: {
        type: String,
        required: true
    },
    address: {
        type: String,
        required: true
    },
    fiscalNumber: {
        type: Number,
        required: true
    },
    email: {
        type: String,
        required: true
    },
    password: {
        type: String,
        required: true
    },
    publicKey: {
        type: String,
        required: true
    },
    cardType: {
        type: String,
        required: true
    },
    cardNumber: {
        type: String,
        required: true
    },
    cardValidity: {
        type: String,
        required: true
    }
});

let Customer = mongoose.model("Customer", customerSchema);

export default Customer;